compiler/record-minmax/README.md

```diff
@@ -0,0 +1,18 @@
+# record-minmax
+
+_record-minmax_ is a tool to embed min/max values of activations to the circle model for post-training quantization.
+
+## Usage
+
+This will run with the path to the input model (.circle), input data (.h5), and the output model (.circle).
+
+```
+$ ./record-minmax <path_to_input_model> <path_to_input_data> <path_to_output_model>
+```
+
+For example,
+```
+$ ./record-minmax input.circle input.h5 out.circle
```

Comment:
HDF5 format has its own benefits, like storing shape etc.
But it is not user friendly, because the user will have to convert their plain data to this HDF5 format. It is far away from user friendliness :)

I whould suggest to make this tool accept plain data (in the same format as fed into the network). 
@jinevening What do you think?

Reply:
> I whould suggest to make this tool accept plain data (in the same format as fed into the network).
@jinevening What do you think?

There are various formats of `plain data`. It can be jpeg, png, or binary-typed files from other frameworks (e.g., tfrecord). Supporting all those formats in the record-minmax executable (c++) would be burdensome.

To improve user friendliness, we're preparing a separate script (maybe written in Python) that converts other data formats (e.g., jpeg, png, or tfrecord) to hdf5.

Reply:
But why hdf5? I can't see benefits of it in this particular case.

Reply:
> It can be jpeg, png, or binary-typed files from other frameworks (e.g., tfrecord). 

I don't agree. The user will not feed jpg or png into the network, all networks expect binary data. Why don't just follow the network input format?

Reply:
>  The user will not feed jpg or png into the network, all networks expect binary data.

Is this some standard? I've no experience in learning stuff. It would help if you give some reference how engineers really do.

What about `Image --> some tool --> convert to binary --> network`
Is this not acceptable?

Reply:
You wrote
> I whould suggest to make this tool accept plain data

I don't get this. first you want to use plain data. now you want to not-plain data.

Reply:
@seanshpark
I would like to use plain data, e.g. pass to the tool the same data as you would pass to the network. I did not say otherwise. If it is "binary data" that confused you, then it is the same thing as "plain data".

> What about Image --> some tool --> convert to binary --> network
Is this not acceptable?

This is OK to me, but I don't understand why ” binary data” in this flow shold be in hdf5 format. Why don't just convert to the format that the network expects?

Reply:
> Why don't just convert to the format that the network expects?

I think we need to figure out about the format that the network expects...

Reply:
> This is OK to me

Just remembered that models may require some input preprocessing. For example, subtract average of all pixel from each pixel before converting them to floats.
This preprocessing is model dependent, and only the user knows what to do for each particular model.
Our scripts that convert png/jpg to floats can support some well known preprocessing (configured via options), but this is not a general solution.
In my opinion therefore, the user must do this preprocessing and convert preprocessed image to the format expected by the model. Quantizer tool shold accept the data in this same format.
The scripts to convert images to plain data could be helpful though, for most popular variants of preprocessing.

Reply:
I'll explain in more details a little later

Reply:
> but this is not a general solution.

I don't think our first target for this year is a general solution.

Reply:
@s-barannikov , so are you going to approve this PR or want to reject this PR?

Reply:
I would like to not use HDF5 format as input, but rather the same format as the network expects.
I will approve the PR it is OK to rework it later. Is it OK?

Reply:
> I will approve the PR it is OK to rework it later. Is it OK?

We can always fix something later. How about firing an issue about the format for discussion?

Reply:
OK)

Reply:
@s-barannikov , perhaps you may make an issue, but I leave this comment.

> This is OK to me, but I don't understand why ” binary data” in this flow shold be in hdf5 format. Why don't just convert to the format that the network expects?

The main reason I chose hdf5 was that it is a well-defined, widely accepted solution to save data; I think this is one of the most persuasive reasons to determine the format of input data. In addition, hdf5 has some practical benefits, e.g., storing the shape as you mentioned earlier, and reusing existing libraries to save and load the data.

> In my opinion therefore, the user must do this preprocessing and convert preprocessed image to the format expected by the model. Quantizer tool shold accept the data in this same format.

I think we can still use hdf5 to save the pre-processed data; hdf5 is just a library for processing and saving data. I'm not 100% sure, but "the format expected by the model" would be able to be expressed in the hdf5 file (Do you know any counter-example ?). 

To use our own file format, some additional works will be required, e.g., determining the layout of the file (use of separate file for data description? how to save the shape and data?), and implementing functions to serialize, deserialize, and iterate the data. I prefer not to increase the amount of our works, unless our own file format has a clear benefit over hdf5.